include/flash_map.h

```diff
@@ -42,6 +42,7 @@ struct flash_area {
 	u16_t pad16;
 	off_t fa_off;
 	size_t fa_size;
+	const char *fa_name;
```

Comment:
I believe it fa_device_id is not needed anymore

Reply:
fa_name -> fa_dev_name

Reply:
> I believe it fa_device_id is not needed anymore

Its not, was keeping it around until we decide on deprecation policy vs removal for out of tree flash_maps.